### ISSUE TITLE
IP Restricted Logins

### ISSUE THREAD

**Issue by iammattmartin** [chatwoot/chatwoot#1101]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
Similar to #486, it would be good for security to limit logins to specific IP addresses or even the admin interface entirely.

**Comment by iammattmartin** [comment]
Maybe to add to this, we have locked down some URLs from non-trusted IPs:

- / (index)
- /app
- /super_admin

We're allowing:

- /* (anything else)

It would be good to know any locations which are admin only, i.e such as for mobile app if they differ to the above list.

**Comment by pranavrajs** [comment]
@sojan-official I did some research on this. This seems to be a straightforward task, we need to define an env variable for allowed IP address.

More information: https://guides.rubyonrails.org/routing.html#advanced-constraints 

@iammattmartin The list you provided seems good to me. I guess you can add the API endpoints also to the list. Let me get back to you on what APIs need to be on the allowed list and what all can be put behind the IP filtering.

**Comment by spanjeta** [comment]
Yes. Agree. Admin panel should have settings for safe IP and restricted url

**Comment by hiagodotme** [comment]
To solve I created a rule in NGINX if it helps in any way, follow the rule.
```nginx
location ^~ /super_admin {
    allow 1.2.3.4;
    deny all;
    proxy_pass http://CHATWOOT_INTERNAL_INSTANCE_IP:3002;
}
```

**Comment by iammattmartin** [comment]
This is effectively what has already been discussed but the thread is more about adding native support in the UI instead of hard-set support in the proxy.
